tests/unit/test_encoder.py

```diff
@@ -76,3 +76,27 @@ def dist(a, b):
     assert small_distance < dist('C', 'B')
     assert small_distance < dist('C', 'A')
     assert small_distance < dist('B', 'A')
+
+
+@pytest.mark.parametrize(
+    ['docs', 'docs_per_path', 'traversal_path'],
+    [
+        (pytest.lazy_fixture('docs_with_text'), [['r', 10], ['c', 0], ['cc', 0]], 'r'),
+        (pytest.lazy_fixture('docs_with_chunk_text'), [['r', 0], ['c', 10], ['cc', 0]], 'c'),
+        (pytest.lazy_fixture('docs_with_chunk_chunk_text'), [['r', 0], ['c', 0], ['cc', 10]], 'cc')
+    ]
+)
+def test_traversal_path(docs: DocumentArray, docs_per_path: List[List[str]], traversal_path: str):
+    def validate_traversal(expected_docs_per_path: List[List[str]]):
+        def validate(res):
+            for path, count in expected_docs_per_path:
+                embeddings = DocumentArray(res).traverse_flat([path]).get_attributes('embedding')
+                for emb in embeddings:
```

Comment:
assert len(embeddings) is of some value. Otherwise empty list of embeddings would pass the test

Reply:
This happens in the end of the test. We return the boolean and assert it later on.